### PR TITLE
Add Headless Data Table Support

### LINKED ISSUE #2920: Support Headless Data Table
SLDS has a [Headless Data Table](https://www.lightningdesignsystem.com/components/data-tables/#Headless) variant that is not currently supported in DSR.

To resolve this issue, a new `headless: boolean` prop should be added to `DataTable`. Setting this to `true` should add the `slds-table_header-hidden` class to `table` and the `slds-assistive-text` class to `thead`.

### PR DESCRIPTION
**Pull request by jacobroschen**
closes #2920

### Additional description

This commit adds a new DataTable prop, `isHeadless: boolean`. Setting
this prop to `true` hides the header by applying
`slds-table_header-hidden` to `table` and `slds-assistive-text` to
`thead`.

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [X] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [X] `npm run lint:fix` has been run and linting passes.
* [X] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [X] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [X] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [X] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [X] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

File: components/data-table.d.ts [hunk 1]
```diff
@@ -62,6 +62,14 @@ declare module '@salesforce/design-system-react/components/data-table' {
 		 * Use this if you are creating an advanced table (selectable, sortable, or resizable rows). Columns widths will be truncate based on width and DOM ancestors. See `fixedHeader` to enable horizontal and vertical scrolling.
 		 */
 		fixedLayout?: boolean;
+		/**
+		 * When fixedHeader is true, specifies whether there's more data to be loaded and displays a spinner at the bottom of the table if so.
+		 */
+		hasMore?: boolean,
```
Comment (jacobroschen) [inline components/data-table.d.ts:68]: It looked like this prop was missed when the TypeScript typings were created.

**Comment by jacobroschen** [comment]
@garygong could you review this when you get a few minutes? Thanks!

File: components/data-table.d.ts (hunk 1 above)
Comment (interactivellama) [reply components/data-table.d.ts:68]: The types were added, but they are automatically generated at this time. Please call them out when you find them.

**Comment by interactivellama** [comment]
Thanks @jacobroschen . Please update prop name to isHeadless to conform to https://github.com/salesforce/design-system-react/blob/master/docs/codebase-overview.md#boolean-props-use-is-has-and-can-prefixes-or--able-suffix

**Comment by interactivellama** [comment]
Rest of PR is good.

**Comment by jacobroschen** [comment]
@interactivellama thanks for the quick review. I've changed `headless` to `isHeadless` and `hidden` to `isHidden`.

**Comment by interactivellama** [comment]
@jacobroschen We got the CI working again. Please update the DOM snapshots.

**Comment by jacobroschen** [comment]
@interactivellama Everything should be ready. Thanks again for getting this in.

**Review by interactivellama — APPROVED** [review]

**Pull request merged**